### PR TITLE
Hns test fix

### PR DESCRIPTION
**Pull request by adamnsm1**
Fixed up the branching issues in hns_test, made a new branch hns_test_fix with the same material. Contains unit tests for all of the so-far implemented features of the hierarchical naming scheme, with the exception of get_num_data_parity_blocks because I can't find where that got moved to.

### CODE REVIEW

File: zookeeper/source/zk_nn_client.cc [hunk 1]
```diff
@@ -1476,7 +1475,7 @@ u_int64_t ZkNnClient::generate_block_group_id() {
 }
 
 u_int64_t ZkNnClient::get_block_group_id(u_int64_t storage_block_id) {
-    u_int64_t mask = ((1ull << 47) - 1) << 16;  // 47 ones and 16 zeros.
+    u_int64_t mask = (~(0ull) << 16);
```
Comment (riceluxs1t) [inline zookeeper/source/zk_nn_client.cc:1478]: why was this change needed?

Comment (adamnsm1) [reply zookeeper/source/zk_nn_client.cc:1478]: It was giving the wrong value, I think because  1ull << 47 - 1 doesn't give all 1's.
